finetuner/tuner/callback/model_checkpoint.py

```diff
@@ -0,0 +1,141 @@
+import os
+from typing import TYPE_CHECKING, Optional
+import numpy as np
+from jina.logging.logger import JinaLogger
+from finetuner.helper import get_framework
+from .base import BaseCallback
+
+
+if TYPE_CHECKING:
+    from ..base import BaseTuner
+
+
+class ModelCheckpointCallback(BaseCallback):
+    """
+    Callback to save model at every epoch or the best model across all epochs
+    `ModelCheckepointCallback` is used in conjunction with training
+    using `finetuner.fit()`
+
+    A few options this callback provides include:
+    - Whether to only keep the model that has achieved the "best performance" so
+        far, or whether to save the model at the end of every epoch regardless of
+        performance.
+    - Definition of 'best'; which quantity to monitor and whether it should be
+        maximized or minimized.
+    - The frequency it should save at. Currently, the callback supports saving at
+        the end of every epoch, or after a fixed number of training batches.
+    """
+
+    def __init__(
+        self,
+        filepath: str = None,
+        save_best_only: Optional[bool] = False,
+        monitor: Optional[str] = 'val_loss',
+        mode: Optional[str] = 'auto',
+    ):
+        """
+        :param filepath: string or `PathLike`, path to save the model file.
+        :param save_best_only: if `save_best_only=True` only the best model
+        model will be saved according to the quantity monitored
+        :param monitor: if `monitor='loss'` best bodel saved will be according
+        to the training loss, if `monitor='val_loss'` best model saved will be
+        according to the validation loss
+        :param mode: one of {'auto', 'min', 'max'}. If `save_best_only=True`, the
+        decision to overwrite the current save file is made based on either
+        the maximization or the minimization of the monitored quantity.
+        For `val_acc`, this should be `max`, for `val_loss` this should be
+        `min`, etc. In `auto` mode, the mode is set to `max` if the quantities
+        monitored are 'acc' or start with 'fmeasure' and are set to `min` for
+        the rest of the quantities.
```

Comment:
Please indent the descriptions of parameters, so that it is visually clear what parameter they belong to